dirsrvtests/tests/suites/upgrade/upgrade_repl_plugin_test.py

```diff
@@ -0,0 +1,80 @@
+import ldap
+import logging
+import pytest
+import os
+from lib389.topologies import topology_st as topo
+from lib389.dseldif import DSEldif
+from lib389.plugins import Plugins
+
+log = logging.getLogger(__name__)
+
+
+def test_repl_plugin_name_change(topo):
+    """Test that the replication plugin name is updated to the new name at
+    server startup.
+
+    :id: c2a7b7fb-6524-4391-8883-683b6af2a1cf
+    :setup: Standalone Instance
+    :steps:
+        1. Stop Server
+        2. Edit repl plugin in  dse.ldif
+        3. Add repl plugin dependency to retro changelog plugin.
+        4. Start server
+        5. Verify old plugin is not found
+        6. Verify new plugin is found
+        7. Verify plugin dependency was updated in retro changelog plugin
+    :expectedresults:
+        1. Success
+        2. Success
+        3. Success
+        4. Success
+        5. Success
+        6. Success
+        7. Success
+    """
+
+    REPL_PLUGIN_DN = "cn=Multisupplier Replication Plugin,cn=plugins,cn=config"
+    REPL_PLUGIN_NAME = "Multisupplier Replication Plugin"
+    REPL_PLUGIN_INIT_ATTR = "nsslapd-pluginInitfunc"
+    REPL_PLUGIN_INIT_FUNC = "replication_multisupplier_plugin_init"
+    REPL_DEPENDS_ATTR = "nsslapd-plugin-depends-on-named"
+    OLD_REPL_PLUGIN_NAME = "Replication Plugin"
+    OLD_REPL_PLUGIN_DN = "cn=Replication Plugin,cn=plugins,cn=config"
+    RETROCL_PLUGIN_DN = "cn=Retro Changelog Plugin,cn=plugins,cn=config"
+    RETROCL_PLUGIN_NAME = "Retro Changelog Plugin"
+
+    # Stop the server
+    topo.standalone.stop()
+
+    # Edit repl plugin in dse.ldif
+    dse_ldif = DSEldif(topo.standalone)
+    dse_ldif.replace(REPL_PLUGIN_DN, REPL_PLUGIN_INIT_ATTR, "multi_old_init_function")
+    dse_ldif.rename(REPL_PLUGIN_DN, OLD_REPL_PLUGIN_DN)
+
+    # Add dependency for repl plugin in retro changelog plugin
+    dse_ldif.replace(RETROCL_PLUGIN_DN, REPL_DEPENDS_ATTR, OLD_REPL_PLUGIN_NAME)
+
+    # Start the server
+    topo.standalone.start()
+
+    # Verify old plugin is deleted
+    with pytest.raises(ldap.NO_SUCH_OBJECT):
+        Plugins(topo.standalone).get(OLD_REPL_PLUGIN_NAME)
+
+    # Verify repl plugin name was changed to the new name
+    plugin = Plugins(topo.standalone).get(REPL_PLUGIN_NAME)
+    assert plugin is not None
+    assert plugin.get_attr_val_utf8_l(REPL_PLUGIN_INIT_ATTR) == REPL_PLUGIN_INIT_FUNC
+
+    # Verify dependency was updated in retro changelog plugin
+    plugin = Plugins(topo.standalone).get(RETROCL_PLUGIN_NAME)
+    assert plugin is not None
+    assert plugin.get_attr_val_utf8_l(REPL_DEPENDS_ATTR) == REPL_PLUGIN_NAME.lower()
```

Comment:
Minor can you restart and verify on the second restart that it still works? That way we know the upgrades persisted.

Reply:
Done and tested